src/models/server_models.rs

```diff
@@ -19,13 +21,26 @@ pub struct SearchParams {
 /// A named struct which is used to deserialize the cookies fetched from the client side.
 #[allow(dead_code)]
 #[derive(Deserialize)]
-pub struct Cookie<'a> {
+pub struct Cookie {
```

Comment:
Is it really necessary to convert `&'a str` to `String`s? This might be worse for performance

Reply:
Yeah, I was getting annoying borrowed value doesn't live long enough when building a Cookie from other borrowed values.

Reply:
> Yeah, I was getting annoying borrowed value doesn't live long enough when building a Cookie from other borrowed values.

For this I would suggest you need to set the cookie type to `Cookie<'_>` in the `search` function and this should solve the problem. 

Also, for more information on lifetimes in rust I would suggest watching this tutorials:

- https://piped.video/watch?v=juIINGuZyBc
- https://piped.video/watch?v=rAl-9HwD858

Reply:
I switched the Cookie back to using &str instead. I wasn't aware I could use a ref keyword in the match Some value to keep the reference in the correct scope.